Brokerages/Binance/BinanceBrokerage.Utility.cs

```diff
@@ -0,0 +1,177 @@
+        /// <param name="payload">the body of the request</param>
+        /// <returns>a token representing the request params</returns>
+        private string AuthenticationToken(string payload)
+        {
+            using (HMACSHA256 hmac = new HMACSHA256(Encoding.UTF8.GetBytes(ApiSecret)))
+            {
+                return ByteArrayToString(hmac.ComputeHash(Encoding.UTF8.GetBytes(payload)));
+            }
+        }
+
+        /// <summary>
+        /// If an IP address exceeds a certain number of requests per minute
+        /// HTTP 429 return code is used when breaking a request rate limit.
+        /// </summary>
+        /// <param name="request"></param>
+        /// <returns></returns>
+        private IRestResponse ExecuteRestRequest(IRestRequest request)
+        {
+            const int maxAttempts = 10;
+            var attempts = 0;
+            IRestResponse response;
+
+            do
+            {
+                if (!_restRateLimiter.WaitToProceed(TimeSpan.Zero))
+                {
+                    OnMessage(new BrokerageMessageEvent(BrokerageMessageType.Warning, "RateLimit",
+                        "The API request has been rate limited. To avoid this message, please reduce the frequency of API calls."));
+
+                    _restRateLimiter.WaitToProceed();
+                }
+
+                response = RestClient.Execute(request);
+                // 429 status code: Too Many Requests
+            } while (++attempts < maxAttempts && (int)response.StatusCode == 429);
+
+            return response;
+        }
+
+        /// <summary>
+        /// Provides the current tickers price
+        /// </summary>
+        /// <returns></returns>
+        public Messages.PriceTicker[] GetTickers()
+        {
+
+            string endpoint = $"/api/v3/ticker/price";
+            var req = new RestRequest(endpoint, Method.GET);
+            var response = ExecuteRestRequest(req);
+            if (response.StatusCode != HttpStatusCode.OK)
+            {
+                throw new Exception($"BinanceBrokerage.GetTick: request failed: [{(int)response.StatusCode}] {response.StatusDescription}, Content: {response.Content}, ErrorMessage: {response.ErrorMessage}");
+            }
+
+            return JsonConvert.DeserializeObject<Messages.PriceTicker[]>(response.Content);
+        }
+
+
+        private decimal GetTickerPrice(Order order)
+        {
+            var security = _securityProvider.GetSecurity(order.Symbol);
+            var tickerPrice = order.Direction == OrderDirection.Buy ? security.AskPrice : security.BidPrice;
+            if (tickerPrice == 0)
+            {
+                var tickers = GetTickers();
+                var ticker = tickers.FirstOrDefault(t => t.Symbol == _symbolMapper.GetBrokerageSymbol(order.Symbol));
+                if (ticker == null)
+                    throw new NotSupportedException($"BinanceBrokerage: Unable to resolve currency conversion pair: {order.Symbol}");
```

Comment:
`KeyNotFoundException` may be better suited here.

Reply:
I've checked other brokerages, and they just throw `Exception`  in that cases. I've changed to `KeyNotFoundException`